tests/installation/addon_products_sle.pm

```diff
@@ -60,20 +60,37 @@ sub handle_all_packages_medium {
     # Record the addons to be enabled for debugging
     record_info 'Extension and Module Selection', join(' ', @addons);
     # Enable the extentions or modules
-    foreach (@addons) {
+    # Also record the addons which require license agreement
+    my @addons_with_license = qw(ha we);
+    my @addons_license_tags = ();
+    for my $a (@addons) {
+        push @addons_license_tags, "addon-license-$a" if grep(/^$a$/, @addons_with_license);
         send_key 'home';
-        send_key_until_needlematch "addon-products-all_packages-$_-highlighted", 'down';
+        send_key_until_needlematch "addon-products-all_packages-$a-highlighted", 'down';
         send_key 'spc';
     }
     send_key $cmd{next};
-    # Confirm all required addons are properly added
-    assert_screen([qw(addon-products sle-product-license-agreement)], 60);
-    if (match_has_tag 'sle-product-license-agreement') {
-        record_soft_failure 'bsc#1081647';
-        wait_screen_change { send_key 'alt-a' };
-        send_key 'alt-n';
-        assert_screen 'addon-products';
+    # Check the addon license agreement
+    my $counter = 2 + (scalar @addons_license_tags);
```

Comment:
I understand why you set this counter number, better to add short comment to describe it :)

Reply:
Added comment to describe why $counter is defined as is